### PR TITLE
Use CSV filename to create IDs

### LINKED ISSUE #35: Identify source file in ID field
Openaddress IDs as stored in our Elasticsearch index are currently just an autoincrementing integer, which isn't very helpful to OA data users and may not be unique across our different sources. Ids should change to have the source file and row in that source file identified in the ID.

### PR DESCRIPTION
**Pull request by orangejulius**
This PR adds a section to generated IDs that is based on the filename where the record came from. The integer part of the ID is still just an incrementing integer that starts at 0.

For example, the first valid document in `us/ny/city_of_new_york.csv` will now have the id `us/ny/city_of_new_york:0`.

Fixes #35

### CODE REVIEW

**Comment by dianashk** [comment]
For now this will work. Eventually, maybe we would want the entire path to the file stored in the id `us/pa/philladelphia.csv` would result in `us-pa-philadelphia:1234`

File: lib/streams/recordStream.js [hunk 1]
```diff
@@ -8,6 +9,29 @@ var CleanupStream = require('./cleanupStream');
...
+ * Construct a suitable id prefix for a CSV file given
+ * its full filename and the base directory of all OA CSV files.
+ */
+function getIdPrefix(filename, dirPath) {
+  var heirarchy = [];
+  var current_level = filename;
+
+  // traverse the directory structure upward to the OA base directory,
+  // keeping track of directory names along the way
+  while (current_level !== '/') {
```
Comment (dianashk) [inline lib/streams/recordStream.js]: this whole bit seems hard to follow. Maybe using something like this would be clearer?

`'/foo/bar/baz.csv'.split(require('path').sep);` results in `[ '', 'foo', 'bar', 'baz.csv' ]`
and then just iterate through the array

Comment (orangejulius) [inline lib/streams/recordStream.js]: I'm normally not a fan of iterative code like this, but the [split based approach](https://github.com/pelias/openaddresses/blob/use_filename_in_id_alternate/lib/streams/recordStream.js#L17) doesn't read much better in my opinion. I'm open to suggestions to make either method cleaner.

Comment (dianashk) [inline lib/streams/recordStream.js]: indeed, maybe something like...

```javascript
var interestingPart = _.replace(filename, dirPath, '');
interestingPart = _.replace(interestingPart, '.csv', '');
interestingPart = _.trim(interestingPart, '/');
return interestingPart;
```

Comment (orangejulius) [inline lib/streams/recordStream.js]: Ok, I like that. It definitely gives up any cross platform support, but we can come back and address that later (I'm sure it's not the only cross platform issue). I've updated this PR with code adapted from yours.

**Comment by dianashk** [comment]
:+1:
